From ff03a5960c24e1939988543e17c8ee2e042e968a Mon Sep 17 00:00:00 2001
From: Vitaly Zaitsev <vitaly@easycoding.org>
Date: Thu, 6 Aug 2020 18:45:32 +0200
Subject: [PATCH] Fixed compatibility issues with old Cmake versions.

Signed-off-by: Vitaly Zaitsev <vitaly@easycoding.org>
---
 CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 45849d4..7cebbc2 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -130,6 +130,7 @@ if (SCN_INSTALL)
     install(TARGETS ${SCN_EXPORT_TARGETS_LIST}
         EXPORT scnTargets
         ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+        LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
         INCLUDES DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
 
     install(EXPORT scnTargets
