From d5b021ff7678e185ecfc748e7f05460265ab7526 Mon Sep 17 00:00:00 2001
From: Vitaly Zaitsev <vitaly@easycoding.org>
Date: Tue, 4 Aug 2020 11:25:25 +0200
Subject: [PATCH] Added SOVERSION field for shared library.

Signed-off-by: Vitaly Zaitsev <vitaly@easycoding.org>
---
 CMakeLists.txt | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 45849d4..978f618 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -78,6 +78,8 @@ set(SCN_EXPORT_TARGETS_LIST scn scn-header-only)
 add_library(scn::scn ALIAS scn)
 add_library(scn::scn-header-only ALIAS scn-header-only)
 
+set_property(TARGET scn PROPERTY SOVERSION 0)
+
 if (SCN_TESTS)
     enable_testing()
     add_subdirectory(test)
